logger.go

```diff
@@ -0,0 +1,175 @@
+package glog
+
+import (
+	"context"
+	"fmt"
+)
+
+// Logger provides logging functionality with additional data and prefixing.
+type Logger struct {
+	*loggingT
+	prefix string
+	// Extra arguments to be appended to each request
+	data []interface{}
+}
+
+// NewLogger creates a Logger instance with no additional data.
+func NewLogger() Logger {
+	return Logger{
+		loggingT: &logging,
+	}
+}
+
+// WithContext creates a logger from a context.Context
+func WithContext(ctx context.Context) Logger {
+	data, _ := ctx.Value(contextKeyData).([]interface{})
+	prefix, _ := ctx.Value(contextKeyPrefix).(string)
+	return Logger{
+		loggingT: &logging,
+		data:     data,
+		prefix:   prefix,
+	}
+}
+
+// WithPrefix creates a Logger with a given prefix.
+func WithPrefix(prefix string) Logger {
+	return Logger{
+		loggingT: &logging,
+		prefix:   prefix,
+	}
+}
+
+// WithData creates a Logger with a given set of backend data.
+// All args provided will be wrapped with glog.Data prior to being sent.
```

Comment:
I don't think I saw where they all get converted to data.

Reply:
Line 168 in the extend func.